test/distributed/fsdp/test_fsdp_param_exec_order_wrap.py

```diff
@@ -0,0 +1,86 @@
+# Owner(s): ["oncall: distributed"]
+
+import torch
+from torch.testing._internal.common_fsdp import FSDPTest
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.distributed.fsdp.wrap import ParamExecOrderWrapPolicy, always_wrap_policy
+from torch.distributed.fsdp.fully_sharded_data_parallel import ShardingStrategy
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+from torch.testing._internal.common_utils import (
+    instantiate_parametrized_tests,
+    parametrize,
+    run_tests,
+)
+
+
+class Model(torch.nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+        self.layer0 = torch.nn.Linear(6, 6)
+        self.layer1 = torch.nn.Linear(6, 6, bias=False)
+        self.layer2 = torch.nn.Sequential(
+            torch.nn.Linear(6, 3, bias=False),
+            torch.nn.ReLU(),
+            torch.nn.Linear(3, 6, bias=False),
+        )
+        self.relu = torch.nn.ReLU()
+
+    def forward(self, x):
+        # `layer0` -> `layer2` -> `layer1`
+        # the forward execution order is NOT consistent with the model definition order.
+        z = self.relu(self.layer0(x))
+        z = self.relu(self.layer2(z))
+        z = self.relu(self.layer1(z))
+        return z
+
+    def get_input(self, device: torch.device):
+        return torch.randn((8, 6)).to(device)
```

Comment:
FYI: I remember seeing in your initial commit(s) that this used to be a `tuple` like `(torch.randn((8, 6)).to(device),)`. I had `get_input()` return a tuple to match the signature from some of the models in `common_fsdp.py` (and have methods like `run_backward()`) so that I could write generic training loops independent of the model.

Reply:
Sounds good, done